test/test_sample.py

```diff
@@ -1795,6 +1795,71 @@ def test_attrs(self):
         with self.assertRaisesRegex(AttributeError, 'Unknown variable name:'):
             dummy = fit.c
 
+    # pylint: disable=use-a-generator
+    def test_diagnostics(self):
+        # centered 8 schools hits funnel
+        stan = os.path.join(DATAFILES_PATH, 'eight_schools.stan')
+        model = CmdStanModel(stan_file=stan)
+        rdata = os.path.join(DATAFILES_PATH, 'eight_schools.data.R')
+        with LogCapture() as log:
+            logging.getLogger()
+            fit = model.sample(
+                data=rdata,
+                seed=55157,
+                show_progress=False,
+                show_console=False,
+            )
+
+            self.assertTrue(
+                all([a == b for a, b in zip(fit.max_treedepths, [0, 0, 0, 6])])
+            )
+            self.assertTrue(
+                all([a == b for a, b in zip(fit.divergences, [10, 143, 5, 4])])
+            )
```

Comment:
can we use `self.assertListEqual(fit.max_treedepths.as_list(), [0,0,0,6])` or something similar?

I also worry that this test will be a bit brittle in terms of cmdstan changes. Maybe we should instead assert that `fit.max_treedepths.sum()` is greater than some small number?

Reply:
indeed it was brittle.  fixed.